framework/controller/controller.gotext

```diff
@@ -69,6 +69,19 @@ func ({{$action.Short}} *{{ $.Pascal }}{{$action.Pascal}}Action) handler(httpRes
 			JSON: response.Status(400).Set("Content-Type", "application/json").JSON(map[string]string{"error": err.Error()}),
 		}
 	}
+	// validate input
+	{{- if (index $action.Params 0).HasValidate }}
```

Comment:
I've been trying to avoid adding logic like into the templates because the templates are difficult to understand as-is. 

I'd create a method on the state that could be used, so it's more like `{{- if $action.Params.HasValidate }}`.

I'm not sure if action params has this already, but I'm pretty sure the action results does already, so it would be a similar change here.